models/rydberg_calcs.py

```diff
@@ -93,6 +127,106 @@ def get_E_TransitionFreq(self):
                                        A=cs().getHFSCoefficients(n=self.n2,
                                                                  l=self.l2, j=self.j2)[0])
         return freq_1 - HFS_g + HFS_e  # in Hz
+    
+
+    def freq_hfs(self, atom, n1, l1, j1, f1, n2, l2, j2, f2, debug = False):
+        """
+        Calculate the hyperfine structure transition frequency between two atomic states.
+
+        This function computes the frequency of a transition between two hyperfine states
+        of an atom, taking into account the hyperfine structure corrections.
+
+        Parameters:
+        atom : Atom object
+            The atom for which the transition is being calculated.
+        n1, l1, j1, f1 : int
+            Quantum numbers for the initial state (principal, orbital, total angular momentum, total atomic angular momentum).
+        n2, l2, j2, f2 : int
+            Quantum numbers for the final state.
+        debug : bool, optional
+            If True, print the calculated frequency in GHz. Default is False.
+
+        Returns:
+        float
+            The hyperfine structure transition frequency in Hz.
+
+        Note:
+        This function uses the `hfs` function to calculate hyperfine structure corrections
+        and the `getTransitionFrequency` method of the atom object to get the base transition frequency.
+        """
+
+
+        Ehfs_1 = self.hfs(atom, n1, l1, j1, f1)
+        Ehfs_2 = self.hfs(atom, n2, l2, j2, f2)
+        f = atom.getTransitionFrequency(n1, l1, j1, n2, l2, j2)
+        f_hfs = f - Ehfs_1 + Ehfs_2
+
+        if debug == True:
+            print(f"Ehfs = {f_hfs/1e9} GHz")
+
+        return f_hfs
+
+
+    def wavelength_hfs(self, atom, n1, l1, j1, f1, n2, l2, j2, f2, debug = False):
+        """
+        return the hyperfine splitting of two states
+
+        Parameters:
+            atom: atom species
+            n1, n2 (int): principles quantum number
+            l1, l2 (int): principles angular momentum number
+            j1, j2 (float): principle total angular momentum (l+s) usually half integer
+        """
+        self.n1 = n1
+        self.l1 = l1
+        self.j1 = j1
+        f = self.freq_hfs(atom, n1, l1, j1, f1, n2, l2, j2, f2, debug = False)
+        w = freq2wavelength(f)
+
+        if debug == True:
+            print(f"λhfs = {w*1e9} nm")
+
+        return w
+
+    def hfs(self, atom, n, l, j, f):
+        A, B = atom.getHFSCoefficients(n, l, j)
+        Ehfs = atom.getHFSEnergyShift(j, f, A, B)
+        return Ehfs
+
+    def alkaline_wavelength_hfs(self, atom, f1 = 3, f2 = 3, transition = "D2", debug = False):
```

Comment:
This function has nothing in particular to do with Rydberg transitions, and would be better placed into a separate class.